platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/ExecutionEngine.java

```diff
@@ -96,14 +99,29 @@ interface Execution {
          */
         // TODO Parametrize UnitOfWork with this generated result
         @Nullable
-        Object getOutput();
+        Object getOutput(File workspace);
```

Comment:
❓ Could we store the workspace location in the `Execution`object (or probably in the `WorkOutput`) and keep `getOutput()` without a parameter? 

It's a bit confusing that a caller of `getOutput` has to know where the output location is in the end.

Reply:
The workspace location is not necessarily the same during execution and when we are resolving the results to be used outside of the execution engine. For tasks and incremental transforms the location is the same, but for immutable work it's different.